spec/models/spree/order_spec.rb

```diff
@@ -324,6 +326,30 @@
     end
   end
 
+  context "#resume" do
+    let(:order) { create(:order_with_totals_and_distribution, :completed) }
+
+    before do
+      order.cancel!
+      order.resume!
+    end
+
+    it "should resume the order" do
+      expect(order.state).to eq 'resumed'
+    end
+
+    it "should resume the shipments" do
+      expect(order.shipments.pluck(:state)).to eq ['pending']
+    end
+
+    context "when payment is in void state" do
+      it "should change the state of the payment to checkout" do
+        order.payments.reload
+        expect(order.payments.pluck(:state)).to eq ['checkout']
```

Comment:
I assumed that `pluck` would query the database without cache. But you using `reload` indicates that in-memory data is used, correct? I didn't know Rails was that "clever". :wink:

Reply:
Great. Thanks for that.